x/gov/types/events.go

```diff
@@ -9,19 +9,20 @@ const (
 	EventTypeActiveProposal   = "active_proposal"
 	EventTypeCancelProposal   = "cancel_proposal"
 
-	AttributeKeyProposalResult              = "proposal_result"
-	AttributeKeyVoter                       = "voter"
-	AttributeKeyOption                      = "option"
-	AttributeKeyProposalID                  = "proposal_id"
-	AttributeKeyProposalMessages            = "proposal_messages" // Msg type_urls in the proposal
-	AttributeKeyVotingPeriodStart           = "voting_period_start"
-	AttributeKeyProposalLog                 = "proposal_log"                // log of proposal execution
-	AttributeValueProposalDropped           = "proposal_dropped"            // didn't meet min deposit
-	AttributeValueProposalPassed            = "proposal_passed"             // met vote quorum
-	AttributeValueProposalRejected          = "proposal_rejected"           // didn't meet vote quorum
-	AttributeValueExpeditedProposalRejected = "expedited_proposal_rejected" // didn't meet expedited vote quorum
-	AttributeValueProposalFailed            = "proposal_failed"             // error on proposal handler
-	AttributeValueProposalCanceled          = "proposal_canceled"           // error on proposal handler
+	AttributeKeyProposalResult               = "proposal_result"
+	AttributeKeyVoter                        = "voter"
+	AttributeKeyOption                       = "option"
+	AttributeKeyProposalID                   = "proposal_id"
+	AttributeKeyProposalMessages             = "proposal_messages" // Msg type_urls in the proposal
+	AttributeKeyVotingPeriodStart            = "voting_period_start"
+	AttributeKeyProposalLog                  = "proposal_log"                 // log of proposal execution
+	AttributeValueProposalDropped            = "proposal_dropped"             // didn't meet min deposit
+	AttributeValueProposalPassed             = "proposal_passed"              // met vote quorum
+	AttributeValueProposalRejected           = "proposal_rejected"            // didn't meet vote quorum
+	AttributeValueExpeditedProposalRejected  = "expedited_proposal_rejected"  // didn't meet expedited vote quorum
+	AttributeValueOptimisticProposalRejected = "optimistic_proposal_rejected" // didn't meet optimistic vote quorum
```

Comment:
is having one of these for each type of proposal necessary?

Reply:
Given that we had expedited, I feel like we should do it for the two other types.